File: tags/2.06/AStyleWxTest/ldtp-jx/AStyleWxLDTP.java
import java.io.File;
import com.cobra.ldtp.Ldtp;

public class AStyleWxLDTP
{
	public static void main(String[] args)
	{
		String exepath = "C:\\Users\\jimp\\Projects\\AStyleWx\\build\\vs2013_3.0\\debug\\AStyleWxd.exe";
		String[] noargs = { "" };

		System.out.println("Running C#");
		System.out.println("Creating ldtp");
		Ldtp ldtp = new Ldtp("*AStyleWx");
		if (ldtp == null)
		{
			System.out.println("Cannot create ldtp object");
			System.out.println("The program has terminated!");
			System.exit(1);
		}

		if (!new File(exepath).isFile())
		{
			System.out.println("Cannot find executable: " + exepath);
			System.out.println("The program has terminated!");
			System.exit(1);
		}

		System.out.println("Launching AStyleWx");
		try
		{
			ldtp.launchApp(exepath, noargs);
		}
		catch (Exception e)
		{
			System.out.println("Error in launchapp: " + exepath);
			System.out.println(e.getMessage());
			System.out.println("The program has terminated!");
			System.exit(1);

		}
		if (ldtp.waitTillGuiExist() == 0)
		{
			System.out.println("Error in WaitTillGuiExist");
			System.out.println("The program has terminated!");
			System.exit(1);
		}

		String[] SubMenus = new String[20];
		System.out.println("\nFile");
		SubMenus = ldtp.listSubMenus("mnuFile");
		PrintSubMenus(SubMenus);
		System.out.println("\nEdit");
		SubMenus = ldtp.listSubMenus("mnuEdit");
		PrintSubMenus(SubMenus);
		System.out.println("\nSearch");
		SubMenus = ldtp.listSubMenus("mnuSearch");
		PrintSubMenus(SubMenus);
		System.out.println("\nView");
		SubMenus = ldtp.listSubMenus("mnuView");
		PrintSubMenus(SubMenus);
		System.out.println("\nTools");
		SubMenus = ldtp.listSubMenus("mnuTools");
		PrintSubMenus(SubMenus);
		System.out.println("\nHelp");
		SubMenus = ldtp.listSubMenus("mnuHelp");
		PrintSubMenus(SubMenus);
		System.out.println();

		System.out.println("Closing AStyleWx");
		ldtp.selectMenuItem("mnuFile;mnuExit");
		if (ldtp.waitTillGuiNotExist("frmAStyleWx") == 0)
		{
			System.out.println("Error in WaitTillGuiNotExist: " + exepath);
			System.out.println("The program has terminated!");
			System.exit(1);
		}

		System.out.println("End of program!");
	}

	private static void PrintSubMenus(String[] subMenus)
	{
		for (String subMenu : subMenus)
		{
			if (subMenu != subMenus[0])
				System.out.print(", ");
			System.out.print(subMenu);
		}
		System.out.print("\n");
	}
}

File: tags/2.06/AStyleWxTest/ldtp-px/AStyleWxLDTP.py
#! /usr/bin/python2
""" ON WINDOWS THIS SCRIPT MUST BE RUN WITH ADMINISTRATIVE PRIVLEDGES.
    If this is not done a message will display that "WinLdtpService has stopped working".
"""

# to disable the print statement and use the print() function (version 3 format)
from __future__ import print_function

import os
import sys
# from time import sleep

# global statements to import ldtp --------------------------------------------

# must check for python 2.7 before importing ldtp
version = " {0}.{1}.{2}  ".format(sys.version_info[0],
                                  sys.version_info[1],
                                  sys.version_info[2])
print("python version " + version)
if not sys.version_info[0] == 2 or not sys.version_info[1] == 7:
    print("This script requires version 2.7")
    print("The script has terminated!")
    sys.exit(1)

print("Importing ldtp")
try:
    import ldtp
except:
    print("ldtp was not imported")
    print("This script must be run as an administrator")
    print("The script has terminated!")
    exit(1)

# -----------------------------------------------------------------------------

def main():
    """Main processing function.
    """
    if os.name == "nt":
        exepath = "C:/Users/jimp/Projects/AStyleWx/build/vs2013_3.0/debug/AStyleWxd.exe"
    else:
        exepath = "../../AStyleWx/build/cb-gcc/wx3.0/bin/astylewxd"
    exepath = exepath.replace('/', os.sep)

    #print(sys.path[0])
    os.chdir(sys.path[0])

    if not os.path.exists(exepath):
        print("Cannot find executable: " + exepath)
        print("The script has terminated!")
        sys.exit(1)

    print("Launching AStyleWx")
    try:
        ldtp.launchapp(exepath)
    except ldtp.client_exception.LdtpExecutionError as err:
        print("Error in launchapp: " + exepath)
        print(err)
        print("The script has terminated!")
        sys.exit(1)
    if not ldtp.waittillguiexist('*AStyleWx'):
        print("Error in waittillguiexist")
        print("The script has terminated!")
        sys.exit(1)

    print('\nFile')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuFile'))
    print('\nEdit')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuEdit'))
    print('\nSearch')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuSearch'))
    print('\nView')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuView'))
    print('\nTools')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuTools'))
    print('\nHelp')
    print(ldtp.listsubmenus('frmAStyleWx', 'mnuHelp'))
    print()

    print("Closing AStyleWx")
    ldtp.selectmenuitem('frmAStyleWx', 'mnuFile;mnuExit')
    if not ldtp.waittillguinotexist('*AStyleWx'):
        print("Error in waittillguinotexist")
        print("The script has terminated!")
        sys.exit(1)

# -----------------------------------------------------------------------------

#selectmenuitem('*gedit', 'mnuFile;mnuOpen')

#~ from ldtp import *
#~ launchapp('notepad')
#~ waittillguiexist('*-Notepad')
#~ selectmenuitem('*-Notepad', 'mnuFile;mnuExit')
#~ waittillguinotexist('*-Notepad')



#~ frm = ooldtp.context('*gedit')
#~ frm.waittillguiexist()
#~ txt_field = frm.getchild('txt1')
#~ txt_field.enterstring('Hello world!<return>bye<return>')
#~ ldtp.imagecapture('*gedit', '/tmp/foo.png')
#~ mnu_quit = frm.getchild('mnuQuit')
#~ mnu_quit.selectmenuitem()
#~ alert = ooldtp.context('Question')
#~ alert.waittillguiexist()
#~ btn = alert.getchild('btnClosewithoutSaving')
#~ btn.click()
#~ frm.waittillguinotexist()

# -----------------------------------------------------------------------------

# make the module executable
if __name__ == "__main__":
    main()
    print("End of script!")
    sys.exit(0)

# -----------------------------------------------------------------------------
